src/components/dialogs/field-hook.js

```diff
@@ -0,0 +1,183 @@
+    return {
+        endAdornment: (
+            <InputAdornment
+                position="end"
+                // hack to circumviate centering of adornment
+                // when TextField has variant 'filled' with 'end' position
+                // : classes.adornRightOther
+                //className={classes.adornRightFilled}
+            >
+                {content}
+            </InputAdornment>
+        ),
+    };
+};
+export const useNameField = ({
+    directoryId,
+    elementType,
+    triggerReset,
+    ...props
+}) => {
+    const [error, setError] = useState();
+    const timer = useRef();
+    const intl = useIntl();
+    const [checking, setChecking] = useState(undefined);
+    const [adornment, setAdornment] = useState();
+
+    const updateValidity = useCallback(
+        (name) => {
+            if (name.replace(/ /g, '') !== '') {
+                //If the name is not only white spaces
+                elementExists(directoryId, name, elementType)
+                    .then((data) => {
+                        setError(
+                            data
+                                ? intl.formatMessage({
+                                      id: 'studyNameAlreadyUsed',
+                                  })
+                                : ''
+                        );
+                    })
+                    .catch((error) => {
+                        setError(
+                            intl.formatMessage({
+                                id: 'nameValidityCheckErrorMsg',
+                            }) + error
+                        );
+                    })
+                    .finally(() => {
+                        setChecking(false);
+                    });
+            } else {
+                setError(intl.formatMessage({ id: 'nameEmpty' }));
+                setChecking(false);
+            }
+        },
+        [directoryId, elementType, intl]
+    );
+
+    useEffect(() => {
+        if (checking === undefined) return;
+        if (checking)
+            setAdornment(
+                makeAdornmentEndIcon(<CircularProgress size="1rem" />)
+            );
+        else if (error) setAdornment(undefined);
+        else
+            setAdornment(
+                makeAdornmentEndIcon(<CheckIcon style={{ color: 'green' }} />)
+            );
+    }, [checking, error]);
+
+    const [name, field] = useTextValue({
+        ...props,
+        triggerReset,
+        error: !!error,
+        adornment: adornment,
+    });
+
+    useEffect(() => {
+        if (name === '' && !timer.current) return; // initial render
+        clearTimeout(timer.current);
+        setChecking(true);
+        setError(undefined);
+        timer.current = setTimeout(() => updateValidity(name), 700);
+    }, [name, updateValidity]);
+
+    useEffect(() => {
+        setError(undefined);
+        timer.current = undefined;
+        setChecking(undefined);
+        setAdornment(undefined);
```

Comment:
Do we need to reset adornment ? Seems a lot static for me...

Reply:
adornment is the check when the name is valid

Reply:
HO I see... it makes confusion with adornment which we use for Unit in some forms textFields... Ok then